File: FinalProject/app/src/main/java/com/example/finalproject/service/spell/Component.java
package com.example.finalproject.service.spell;

public class Component {
    private String name;
    private boolean available;

    public Component(String name, boolean available) {
        this.name = name;
        this.available=available;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isAvailable() {
        return available;
    }

    public void setAvailable() {
        available=true;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/service/classes/spell/Component.kt
package com.example.finalproject.service.classes.spell

import com.example.finalproject.service.classes.Unit
import com.example.finalproject.service.serializers.ComponentSerializer
import kotlinx.serialization.Serializable

@Serializable(with= ComponentSerializer::class)
open class Component(name: String, id: Int, var available: Boolean) :
    Unit(name = name, id = id) {

    constructor():this("", -1, false)
    constructor(component: Component):this(component.name, component.id, component.available)

    fun avail() {
        available = true
    }
}